src/executor_factory.py

```diff
@@ -185,35 +201,47 @@ def debug_executor(self, output_path, executor_name, num_approach, packages, des
+
+                if is_dependency_issue:
+                    all_files_string = self.files_to_string({
+                        key: val for key, val in file_name_to_content.items() if key in ['requirements.txt', 'Dockerfile']
+                    })
+                    user_query = (
+                        f"Your task is to provide guidance on how to solve an error that occurred during the Docker "
+                        f"build process. The error message is:\n{error}\nTo solve this error, you should first "
+                        f"identify the type of error by examining the stack trace. Once you have identified the "
+                        f"error, you should suggest how to solve it. Your response should include the files that "
+                        f"need to be changed, but not files that don't need to be changed. For files that need to "
+                        f"be changed, please provide the complete file with the exact same syntax to wrap the code.\n\n"
+                        f"You are given the following files:\n\n{all_files_string}"
+                    )
+                else:
+                    all_files_string = self.files_to_string(file_name_to_content)
+                    user_query = (
+                             f"General rules: " + not_allowed()
+                            + f'Here is the description of the task the executor must solve:\n{description}'
+                            + f'\n\nHere is the test scenario the executor must pass:\n{test}'
+                            + f'Here are all the files I use:\n{all_files_string}'
+                            + f'\n\nThis error happens during the docker build process:\n{error}\n\n'
+                            + ((f'This is an error that is already fixed before:\n{error_before}\n\n') if error_before else '')
+                            + 'Look at exactly at the stack trace. First, think about what kind of error is this? '
+                              'Then think about possible reasons which might have caused it. Then suggest how to '
+                              'solve it. Output the files that need change. '
+                              "Don't output files that don't need change. If you output a file, then write the "
+                              "complete file. Use the exact same syntax to wrap the code:\n"
+                              f"**...**\n"
+                              f"```...\n"
+                              f"...code...\n"
+                              f"```"
+                    )
+
                 conversation = self.gpt_session.get_conversation()
                 returned_files_raw = conversation.query(user_query)
                 for file_name, tag in FILE_AND_TAG_PAIRS:
                     updated_file = self.extract_content_from_result(returned_files_raw, file_name)
-                    if updated_file:
+                    if updated_file and (not is_dependency_issue or file_name in ['requirements.txt', 'Dockerfile']):
```

Comment:
refactor to handle update of code better by always copying all files and only updating the relevant ones

Reply:
this refactoring can be done in a followup pr and is not critical - just better coding style